eth/db/chain.py

```diff
@@ -129,6 +130,34 @@ def persist_block(self,
         with self.db.atomic_batch() as db:
             return self._persist_block(db, block, genesis_parent_hash)
 
+    def persist_unexecuted_block(self,
+                                 block: BlockAPI,
+                                 receipts: Tuple[ReceiptAPI, ...],
+                                 genesis_parent_hash: Hash32 = GENESIS_PARENT_HASH
+                                 ) -> Tuple[Tuple[Hash32, ...], Tuple[Hash32, ...]]:
+        with self.db.atomic_batch() as db:
```

Comment:
I think this can move down to just above the call to `self._persist_tree_data_dict`.  Probably a minor improvement at best but it reduces the amount of time that the batch is open.